failed: EventSourcedBehaviorSpec - adhere default and disabled Recovery strategies

https://github.com/akka/akka/runs/3315690054?check_suite_focus=true#step:5:20416

```
--> [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec: A typed persistent actor must adhere default and disabled Recovery strategies] Start of log messages of test that [Failed(java.lang.AssertionError: expected State(4,Vector(0, 1, 2, 3)), found State(3,Vector(0, 1, 2)))]
20308
2021-08-12 20:45:47,788 [INFO ] [akka.event.slf4j.Slf4jLogger] [] [] [] [] - Slf4jLogger started 
20309
2021-08-12 20:45:47,823 [DEBUG] [EventSourcedBehaviorSpec] [] [] [] [] - Starting ActorTestKit 
20310
2021-08-12 20:45:47,823 [DEBUG] [akka.persistence.testkit.query.scaladsl.PersistenceTestKitReadJournal] [] [] [] [] - Using in memory storage [akka.persistence.testkit.journal] for test kit read journal 
20311
2021-08-12 20:45:47,823 [DEBUG] [akka.persistence.testkit.query.scaladsl.PersistenceTestKitReadJournal] [] [] [] [] - Using in memory storage [akka.persistence.testkit.journal] for test kit read journal 
20312
2021-08-12 20:45:47,837 [INFO ] [akka.actor.testkit.typed.scaladsl.LogCapturing] [] [] [] [] - Logging started for test [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec: A typed persistent actor must persist an event] 
20313
2021-08-12 20:45:47,843 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$a] [get-permit] [c1)] - Initializing snapshot recovery: Recovery(SnapshotSelectionCriteria(9223372036854775807,9223372036854775807,0,0),9223372036854775807,9223372036854775807) 
20314
2021-08-12 20:45:47,845 [DEBUG] [akka.actor.typed.Behavior$] [] [akka://EventSourcedBehaviorSpec/user/$a] [] [] - Snapshot recovered from 0 Map() VersionVector() 
20315
2021-08-12 20:45:47,845 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$a] [replay-evt] [c1)] - Replaying events: from: 1, to: 9223372036854775807 
20316
2021-08-12 20:45:47,845 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$a] [replay-evt] [c1)] - Recovery successful, recovered until sequenceNr: [0] 
20317
2021-08-12 20:45:47,845 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$a] [replay-evt] [c1)] - Returning recovery permit, reason: replay completed successfully 
20318
2021-08-12 20:45:47,845 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$a] [replay-evt] [c1)] - Recovery for persistenceId [PersistenceId(c1))] took 341.0 μs 
20319
2021-08-12 20:45:47,846 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$a] [running-cmd] [c1)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Increment$], resulting effect: [Persist(akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Incremented)], side effects: [0] 
20320
2021-08-12 20:45:47,864 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$a] [persist-evt] [c1)] - Received Journal response: WriteMessagesSuccessful after: 17819831 nanos 
20321
2021-08-12 20:45:47,864 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$a] [persist-evt] [c1)] - Received Journal response: WriteMessageSuccess(PersistentRepr(c1),1,1b05409b-0efc-468c-b7a7-860722808b0e,0,None),151) after: 17919932 nanos 
20322
2021-08-12 20:45:47,865 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$a] [running-cmd] [c1)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$GetValue], resulting effect: [PersistNothing], side effects: [0] 
20323
2021-08-12 20:45:47,865 [INFO ] [akka.actor.testkit.typed.scaladsl.LogCapturing] [] [] [] [] - Logging finished for test [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec: A typed persistent actor must persist an event] that [Succeeded] 
20324
2021-08-12 20:45:47,865 [INFO ] [akka.actor.testkit.typed.scaladsl.LogCapturing] [] [] [] [] - Logging started for test [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec: A typed persistent actor must replay stored events] 
20325
2021-08-12 20:45:47,866 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$b] [get-permit] [c2)] - Initializing snapshot recovery: Recovery(SnapshotSelectionCriteria(9223372036854775807,9223372036854775807,0,0),9223372036854775807,9223372036854775807) 
20326
2021-08-12 20:45:47,867 [DEBUG] [akka.actor.typed.Behavior$] [] [akka://EventSourcedBehaviorSpec/user/$b] [] [] - Snapshot recovered from 0 Map() VersionVector() 
20327
2021-08-12 20:45:47,867 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$b] [replay-evt] [c2)] - Replaying events: from: 1, to: 9223372036854775807 
20328
2021-08-12 20:45:47,867 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$b] [replay-evt] [c2)] - Recovery successful, recovered until sequenceNr: [0] 
20329
2021-08-12 20:45:47,867 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$b] [replay-evt] [c2)] - Returning recovery permit, reason: replay completed successfully 
20330
2021-08-12 20:45:47,867 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$b] [replay-evt] [c2)] - Recovery for persistenceId [PersistenceId(c2))] took 334.5 μs 
20331
2021-08-12 20:45:47,867 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$b] [running-cmd] [c2)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Increment$], resulting effect: [Persist(akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Incremented)], side effects: [0] 
20332
2021-08-12 20:45:47,868 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$b] [persist-evt] [c2)] - Received Journal response: WriteMessagesSuccessful after: 577104 nanos 
20333
2021-08-12 20:45:47,868 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$b] [persist-evt] [c2)] - Received Journal response: WriteMessageSuccess(PersistentRepr(c2),1,745275df-2842-4df6-be64-db9a2d320d5a,0,None),152) after: 621504 nanos 
20334
2021-08-12 20:45:47,868 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$b] [running-cmd] [c2)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Increment$], resulting effect: [Persist(akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Incremented)], side effects: [0] 
20335
2021-08-12 20:45:47,869 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$b] [persist-evt] [c2)] - Received Journal response: WriteMessagesSuccessful after: 296802 nanos 
20336
2021-08-12 20:45:47,869 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$b] [persist-evt] [c2)] - Received Journal response: WriteMessageSuccess(PersistentRepr(c2),2,745275df-2842-4df6-be64-db9a2d320d5a,0,None),152) after: 335202 nanos 
20337
2021-08-12 20:45:47,869 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$b] [running-cmd] [c2)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Increment$], resulting effect: [Persist(akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Incremented)], side effects: [0] 
20338
2021-08-12 20:45:47,869 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$b] [persist-evt] [c2)] - Received Journal response: WriteMessagesSuccessful after: 218002 nanos 
20339
2021-08-12 20:45:47,869 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$b] [persist-evt] [c2)] - Received Journal response: WriteMessageSuccess(PersistentRepr(c2),3,745275df-2842-4df6-be64-db9a2d320d5a,0,None),152) after: 253802 nanos 
20340
2021-08-12 20:45:47,869 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$b] [running-cmd] [c2)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$GetValue], resulting effect: [PersistNothing], side effects: [0] 
20341
2021-08-12 20:45:47,869 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$c] [get-permit] [c2)] - Initializing snapshot recovery: Recovery(SnapshotSelectionCriteria(9223372036854775807,9223372036854775807,0,0),9223372036854775807,9223372036854775807) 
20342
2021-08-12 20:45:47,870 [DEBUG] [akka.actor.typed.Behavior$] [] [akka://EventSourcedBehaviorSpec/user/$c] [] [] - Snapshot recovered from 0 Map() VersionVector() 
20343
2021-08-12 20:45:47,870 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$c] [replay-evt] [c2)] - Replaying events: from: 1, to: 9223372036854775807 
20344
2021-08-12 20:45:47,870 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$c] [replay-evt] [c2)] - Recovery successful, recovered until sequenceNr: [3] 
20345
2021-08-12 20:45:47,870 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$c] [replay-evt] [c2)] - Returning recovery permit, reason: replay completed successfully 
20346
2021-08-12 20:45:47,870 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$c] [replay-evt] [c2)] - Recovery for persistenceId [PersistenceId(c2))] took 537.7 μs 
20347
2021-08-12 20:45:47,871 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$c] [running-cmd] [c2)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$GetValue], resulting effect: [PersistNothing], side effects: [0] 
20348
2021-08-12 20:45:47,871 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$c] [running-cmd] [c2)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Increment$], resulting effect: [Persist(akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Incremented)], side effects: [0] 
20349
2021-08-12 20:45:47,871 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$c] [persist-evt] [c2)] - Received Journal response: WriteMessagesSuccessful after: 707305 nanos 
20350
2021-08-12 20:45:47,871 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$c] [persist-evt] [c2)] - Received Journal response: WriteMessageSuccess(PersistentRepr(c2),4,a19c8cd1-6e1c-4e48-b4a8-fadfe534bea8,0,None),153) after: 752305 nanos 
20351
2021-08-12 20:45:47,871 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$c] [running-cmd] [c2)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$GetValue], resulting effect: [PersistNothing], side effects: [0] 
20352
2021-08-12 20:45:47,872 [INFO ] [akka.actor.testkit.typed.scaladsl.LogCapturing] [] [] [] [] - Logging finished for test [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec: A typed persistent actor must replay stored events] that [Succeeded] 
20353
2021-08-12 20:45:47,872 [INFO ] [akka.actor.testkit.typed.scaladsl.LogCapturing] [] [] [] [] - Logging started for test [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec: A typed persistent actor must handle Terminated signal] 
20354
2021-08-12 20:45:47,873 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$d] [get-permit] [c3)] - Initializing snapshot recovery: Recovery(SnapshotSelectionCriteria(9223372036854775807,9223372036854775807,0,0),9223372036854775807,9223372036854775807) 
20355
2021-08-12 20:45:47,874 [DEBUG] [akka.actor.typed.Behavior$] [] [akka://EventSourcedBehaviorSpec/user/$d] [] [] - Snapshot recovered from 0 Map() VersionVector() 
20356
2021-08-12 20:45:47,874 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$d] [replay-evt] [c3)] - Replaying events: from: 1, to: 9223372036854775807 
20357
2021-08-12 20:45:47,874 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$d] [replay-evt] [c3)] - Recovery successful, recovered until sequenceNr: [0] 
20358
2021-08-12 20:45:47,874 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$d] [replay-evt] [c3)] - Returning recovery permit, reason: replay completed successfully 
20359
2021-08-12 20:45:47,874 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$d] [replay-evt] [c3)] - Recovery for persistenceId [PersistenceId(c3))] took 240.6 μs 
20360
2021-08-12 20:45:47,874 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$d] [running-cmd] [c3)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Increment$], resulting effect: [Persist(akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Incremented)], side effects: [0] 
20361
2021-08-12 20:45:47,874 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$d] [persist-evt] [c3)] - Received Journal response: WriteMessagesSuccessful after: 264802 nanos 
20362
2021-08-12 20:45:47,874 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$d] [persist-evt] [c3)] - Received Journal response: WriteMessageSuccess(PersistentRepr(c3),1,0505550e-e137-4e6d-8a39-485d3669e9a9,0,None),154) after: 302103 nanos 
20363
2021-08-12 20:45:47,875 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$d] [running-cmd] [c3)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$IncrementLater$], resulting effect: [PersistNothing], side effects: [0] 
20364
2021-08-12 20:45:47,875 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$d] [running-cmd] [c3)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$GetValue], resulting effect: [PersistNothing], side effects: [0] 
20365
2021-08-12 20:45:47,886 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$d] [running-cmd] [c3)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$GetValue], resulting effect: [PersistNothing], side effects: [0] 
20366
2021-08-12 20:45:47,897 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$d] [running-cmd] [c3)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$GetValue], resulting effect: [PersistNothing], side effects: [0] 
20367
2021-08-12 20:45:47,899 [DEBUG] [akka.actor.ActorCell] [] [akka://EventSourcedBehaviorSpec/user/$d/$a] [] [] - Cancel all timers 
20368
2021-08-12 20:45:47,899 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$d] [running-cmd] [c3)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$DelayFinished$], resulting effect: [Persist(akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Incremented)], side effects: [0] 
20369
2021-08-12 20:45:47,899 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$d] [persist-evt] [c3)] - Received Journal response: WriteMessagesSuccessful after: 277202 nanos 
20370
2021-08-12 20:45:47,899 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$d] [persist-evt] [c3)] - Received Journal response: WriteMessageSuccess(PersistentRepr(c3),2,0505550e-e137-4e6d-8a39-485d3669e9a9,0,None),154) after: 300802 nanos 
20371
2021-08-12 20:45:47,907 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$d] [running-cmd] [c3)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$GetValue], resulting effect: [PersistNothing], side effects: [0] 
20372
2021-08-12 20:45:47,907 [INFO ] [akka.actor.testkit.typed.scaladsl.LogCapturing] [] [] [] [] - Logging finished for test [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec: A typed persistent actor must handle Terminated signal] that [Succeeded] 
20373
2021-08-12 20:45:47,907 [INFO ] [akka.actor.testkit.typed.scaladsl.LogCapturing] [] [] [] [] - Logging started for test [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec: A typed persistent actor must handle receive timeout] 
20374
2021-08-12 20:45:47,908 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$e] [get-permit] [c4)] - Initializing snapshot recovery: Recovery(SnapshotSelectionCriteria(9223372036854775807,9223372036854775807,0,0),9223372036854775807,9223372036854775807) 
20375
2021-08-12 20:45:47,909 [DEBUG] [akka.actor.typed.Behavior$] [] [akka://EventSourcedBehaviorSpec/user/$e] [] [] - Snapshot recovered from 0 Map() VersionVector() 
20376
2021-08-12 20:45:47,909 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$e] [replay-evt] [c4)] - Replaying events: from: 1, to: 9223372036854775807 
20377
2021-08-12 20:45:47,910 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$e] [replay-evt] [c4)] - Recovery successful, recovered until sequenceNr: [0] 
20378
2021-08-12 20:45:47,910 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$e] [replay-evt] [c4)] - Returning recovery permit, reason: replay completed successfully 
20379
2021-08-12 20:45:47,910 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$e] [replay-evt] [c4)] - Recovery for persistenceId [PersistenceId(c4))] took 1.181 ms 
20380
2021-08-12 20:45:47,910 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$e] [running-cmd] [c4)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Increment$], resulting effect: [Persist(akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Incremented)], side effects: [0] 
20381
2021-08-12 20:45:47,911 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$e] [persist-evt] [c4)] - Received Journal response: WriteMessagesSuccessful after: 424103 nanos 
20382
2021-08-12 20:45:47,911 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$e] [persist-evt] [c4)] - Received Journal response: WriteMessageSuccess(PersistentRepr(c4),1,bac273b4-1481-4fa9-9bb8-d3c0b9472e51,0,None),155) after: 456203 nanos 
20383
2021-08-12 20:45:47,912 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$e] [running-cmd] [c4)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$IncrementAfterReceiveTimeout$], resulting effect: [PersistNothing], side effects: [0] 
20384
2021-08-12 20:45:47,940 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$e] [running-cmd] [c4)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Timeout$], resulting effect: [Persist(akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Incremented)], side effects: [0] 
20385
2021-08-12 20:45:47,940 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$e] [persist-evt] [c4)] - Received Journal response: WriteMessagesSuccessful after: 734206 nanos 
20386
2021-08-12 20:45:47,940 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$e] [persist-evt] [c4)] - Received Journal response: WriteMessageSuccess(PersistentRepr(c4),2,bac273b4-1481-4fa9-9bb8-d3c0b9472e51,0,None),155) after: 820607 nanos 
20387
2021-08-12 20:45:48,409 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$e] [running-cmd] [c4)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$GetValue], resulting effect: [PersistNothing], side effects: [0] 
20388
2021-08-12 20:45:48,410 [INFO ] [akka.actor.testkit.typed.scaladsl.LogCapturing] [] [] [] [] - Logging finished for test [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec: A typed persistent actor must handle receive timeout] that [Succeeded] 
20389
2021-08-12 20:45:48,410 [INFO ] [akka.actor.testkit.typed.scaladsl.LogCapturing] [] [] [] [] - Logging started for test [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec: A typed persistent actor must adhere default and disabled Recovery strategies] 
20390
2021-08-12 20:45:48,413 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$f] [get-permit] [c5)] - Initializing snapshot recovery: Recovery(SnapshotSelectionCriteria(9223372036854775807,9223372036854775807,0,0),9223372036854775807,9223372036854775807) 
20391
2021-08-12 20:45:48,413 [DEBUG] [akka.actor.typed.Behavior$] [] [akka://EventSourcedBehaviorSpec/user/$f] [] [] - Snapshot recovered from 0 Map() VersionVector() 
20392
2021-08-12 20:45:48,413 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$f] [replay-evt] [c5)] - Replaying events: from: 1, to: 9223372036854775807 
20393
2021-08-12 20:45:48,413 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$f] [replay-evt] [c5)] - Recovery successful, recovered until sequenceNr: [0] 
20394
2021-08-12 20:45:48,413 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$f] [replay-evt] [c5)] - Returning recovery permit, reason: replay completed successfully 
20395
2021-08-12 20:45:48,414 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$f] [replay-evt] [c5)] - Recovery for persistenceId [PersistenceId(c5))] took 352.1 μs 
20396
2021-08-12 20:45:48,414 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$f] [running-cmd] [c5)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Increment$], resulting effect: [Persist(akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Incremented)], side effects: [0] 
20397
2021-08-12 20:45:48,414 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$f] [persist-evt] [c5)] - Received Journal response: WriteMessagesSuccessful after: 31600 nanos 
20398
2021-08-12 20:45:48,414 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$f] [persist-evt] [c5)] - Received Journal response: WriteMessageSuccess(PersistentRepr(c5),1,db568ac5-8e99-4db0-ac25-6399c3593257,0,None),156) after: 47200 nanos 
20399
2021-08-12 20:45:48,414 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$f] [running-cmd] [c5)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Increment$], resulting effect: [Persist(akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Incremented)], side effects: [0] 
20400
2021-08-12 20:45:48,414 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$f] [persist-evt] [c5)] - Received Journal response: WriteMessagesSuccessful after: 9500 nanos 
20401
2021-08-12 20:45:48,414 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$f] [persist-evt] [c5)] - Received Journal response: WriteMessageSuccess(PersistentRepr(c5),2,db568ac5-8e99-4db0-ac25-6399c3593257,0,None),156) after: 32500 nanos 
20402
2021-08-12 20:45:48,414 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$f] [running-cmd] [c5)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Increment$], resulting effect: [Persist(akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Incremented)], side effects: [0] 
20403
2021-08-12 20:45:48,415 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$f] [persist-evt] [c5)] - Received Journal response: WriteMessagesSuccessful after: 10000 nanos 
20404
2021-08-12 20:45:48,415 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$f] [persist-evt] [c5)] - Received Journal response: WriteMessageSuccess(PersistentRepr(c5),3,db568ac5-8e99-4db0-ac25-6399c3593257,0,None),156) after: 22301 nanos 
20405
2021-08-12 20:45:48,415 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$g] [get-permit] [c5)] - Initializing snapshot recovery: Recovery(SnapshotSelectionCriteria(9223372036854775807,9223372036854775807,0,0),9223372036854775807,9223372036854775807) 
20406
2021-08-12 20:45:48,415 [DEBUG] [akka.actor.typed.Behavior$] [] [akka://EventSourcedBehaviorSpec/user/$g] [] [] - Snapshot recovered from 0 Map() VersionVector() 
20407
2021-08-12 20:45:48,416 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$g] [replay-evt] [c5)] - Replaying events: from: 1, to: 9223372036854775807 
20408
2021-08-12 20:45:48,416 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$g] [replay-evt] [c5)] - Recovery successful, recovered until sequenceNr: [3] 
20409
2021-08-12 20:45:48,416 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$f] [running-cmd] [c5)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$GetValue], resulting effect: [PersistNothing], side effects: [0] 
20410
2021-08-12 20:45:48,416 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$g] [replay-evt] [c5)] - Returning recovery permit, reason: replay completed successfully 
20411
2021-08-12 20:45:48,416 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$f] [running-cmd] [c5)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Increment$], resulting effect: [Persist(akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$Incremented)], side effects: [0] 
20412
2021-08-12 20:45:48,416 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$g] [replay-evt] [c5)] - Recovery for persistenceId [PersistenceId(c5))] took 500.4 μs 
20413
2021-08-12 20:45:48,416 [DEBUG] [akka.persistence.typed.internal.EventSourcedBehaviorImpl] [] [akka://EventSourcedBehaviorSpec/user/$g] [running-cmd] [c5)] - Handled command [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec$GetValue], resulting effect: [PersistNothing], side effects: [0] 
20414
2021-08-12 20:45:48,417 [INFO ] [akka.actor.testkit.typed.scaladsl.LogCapturing] [] [] [] [] - Logging finished for test [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec: A typed persistent actor must adhere default and disabled Recovery strategies] that [Failed(java.lang.AssertionError: expected State(4,Vector(0, 1, 2, 3)), found State(3,Vector(0, 1, 2)))] 
20415
<-- [akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec: A typed persistent actor must adhere default and disabled Recovery strategies] End of log messages of test that [Failed(java.lang.AssertionError: expected State(4,Vector(0, 1, 2, 3)), found State(3,Vector(0, 1, 2)))]
20416
[info] - must adhere default and disabled Recovery strategies *** FAILED *** (23 milliseconds)
20417
[info]   java.lang.AssertionError: expected State(4,Vector(0, 1, 2, 3)), found State(3,Vector(0, 1, 2))
20418
[info]   at akka.actor.testkit.typed.internal.TestProbeImpl.assertFail(TestProbeImpl.scala:399)
20419
[info]   at akka.actor.testkit.typed.internal.TestProbeImpl.expectMessage_internal(TestProbeImpl.scala:170)
20420
[info]   at akka.actor.testkit.typed.internal.TestProbeImpl.expectMessage(TestProbeImpl.scala:149)
20421
[info]   at akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec.$anonfun$new$8(EventSourcedBehaviorSpec.scala:364)
20422
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
20423
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
20424
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
20425
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
20426
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
20427
[info]   at org.scalatest.wordspec.AnyWordSpecLike$$anon$3.apply(AnyWordSpecLike.scala:1076)
20428
[info]   at akka.actor.testkit.typed.scaladsl.LogCapturing.withFixture(LogCapturing.scala:70)
20429
[info]   at akka.actor.testkit.typed.scaladsl.LogCapturing.withFixture$(LogCapturing.scala:68)
20430
[info]   at akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec.withFixture(EventSourcedBehaviorSpec.scala:280)
20431
[info]   at org.scalatest.wordspec.AnyWordSpecLike.invokeWithFixture$1(AnyWordSpecLike.scala:1074)
20432
[info]   at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$runTest$1(AnyWordSpecLike.scala:1086)
20433
[info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
20434
[info]   at org.scalatest.wordspec.AnyWordSpecLike.runTest(AnyWordSpecLike.scala:1086)
20435
[info]   at org.scalatest.wordspec.AnyWordSpecLike.runTest$(AnyWordSpecLike.scala:1068)
20436
[info]   at akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec.runTest(EventSourcedBehaviorSpec.scala:280)
20437
[info]   at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$runTests$1(AnyWordSpecLike.scala:1145)
20438
[info]   at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:413)
20439
[info]   at scala.collection.immutable.List.foreach(List.scala:333)
20440
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
20441
[info]   at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:390)
20442
[info]   at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:427)
20443
[info]   at scala.collection.immutable.List.foreach(List.scala:333)
20444
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
20445
[info]   at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:396)
20446
[info]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:475)
20447
[info]   at org.scalatest.wordspec.AnyWordSpecLike.runTests(AnyWordSpecLike.scala:1145)
20448
[info]   at org.scalatest.wordspec.AnyWordSpecLike.runTests$(AnyWordSpecLike.scala:1144)
20449
[info]   at akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec.runTests(EventSourcedBehaviorSpec.scala:280)
20450
[info]   at org.scalatest.Suite.run(Suite.scala:1112)
20451
[info]   at org.scalatest.Suite.run$(Suite.scala:1094)
20452
[info]   at akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit.org$scalatest$BeforeAndAfterAll$$super$run(ScalaTestWithActorTestKit.scala:31)
20453
[info]   at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:213)
20454
[info]   at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)
20455
[info]   at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
20456
[info]   at akka.persistence.typed.scaladsl.EventSourcedBehaviorSpec.org$scalatest$wordspec$AnyWordSpecLike$$super$run(EventSourcedBehaviorSpec.scala:280)
20457
[info]   at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$run$1(AnyWordSpecLike.scala:1190)
```

--- a/akka-persistence-typed-tests/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorSpec.scala
+++ b/akka-persistence-typed-tests/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorSpec.scala
@@ -357,15 +357,15 @@ class EventSourcedBehaviorSpec
       counterSetup ! Increment
       counterSetup ! GetValue(probe.ref)
       probe.expectMessage(State(3, Vector(0, 1, 2)))
-
-      val counterDefaultRecoveryStrategy = spawn(counterWithRecoveryStrategy(Recovery.default))
       counterSetup ! Increment
-
       counterSetup ! StopIt
       probe.expectTerminated(counterSetup)
 
+      val counterDefaultRecoveryStrategy = spawn(counterWithRecoveryStrategy(Recovery.default))
       counterDefaultRecoveryStrategy ! GetValue(probe.ref)
       probe.expectMessage(State(4, Vector(0, 1, 2, 3)))
+      counterDefaultRecoveryStrategy ! StopIt
+      probe.expectTerminated(counterDefaultRecoveryStrategy)
 
       val counterDisabledRecoveryStrategy = spawn(counterWithRecoveryStrategy(Recovery.disabled))
       counterDisabledRecoveryStrategy ! Increment
@@ -393,9 +393,9 @@ class EventSourcedBehaviorSpec
       eventProbe.receiveMessages(3)
       snapshotProbe.receiveMessages(3)
       counterSetup ! GetValue(commandProbe.ref)
+      commandProbe.expectMessage(State(3, Vector(0, 1, 2)))
       counterSetup ! StopIt
       commandProbe.expectTerminated(counterSetup)
-      commandProbe.expectMessage(State(3, Vector(0, 1, 2)))
 
       val counterWithSnapshotSelectionCriteriaNone = spawn(
         counterWithSnapshotSelectionCriteria(Recovery.withSnapshotSelectionCriteria(SnapshotSelectionCriteria.none)))

